### PR TITLE
Show embedded projects when a gpkg is dropped onto QGIS

### PR DESCRIPTION
**Pull request by nyalldawson**
Builds off https://github.com/qgis/QGIS/pull/44085, and adds support for showing embedded projects in the dialog when dragging and dropping a gpkg containing projects to QGIS:

image

### CODE REVIEW

**Review by elpaso — APPROVED** [review]

**Comment by nyalldawson** [comment]
@elpaso does that approval include https://github.com/qgis/QGIS/pull/44085?

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -7675,41 +7339,111 @@ bool QgisApp::openLayer( const QString &fileName, bool allowInteractive )
...
   }
 
-  // try to load it as raster
-  if ( QgsRasterLayer::isValidRasterFileName( fileName ) )
+  QList< QgsProviderSublayerModel::NonLayerItem > nonLayerItems;
+  if ( QgsProjectStorage *ps = QgsApplication::projectStorageRegistry()->projectStorageFromUri( fileName ) )
   {
-    // open .adf as a directory
-    if ( fileName.endsWith( QLatin1String( ".adf" ), Qt::CaseInsensitive ) )
+    const QStringList projects = ps->listProjects( fileName );
```
Comment (elpaso) [inline src/app/qgisapp.cpp:7345]: is `ps` guaranteed to be not null?

**Comment by elpaso** [comment]
> @elpaso does that approval include #44085?

Yes, everything looks good to me.

**Comment by m-kuhn** [comment]
This deserves a big :heart:

File: src/app/qgisapp.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/qgisapp.cpp:7345]: Yep, it's checked a couple of lines above

**Pull request merged**